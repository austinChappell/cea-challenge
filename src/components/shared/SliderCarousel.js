import React, { Component } from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  children: PropTypes.arrayOf(PropTypes.array).isRequired,
  title: PropTypes.string,
};

const defaultProps = {
  title: null,
};

class SliderCarousel extends Component {
  state = {}
  render() {
    return (
      <div className="SliderCarousel">
        <h4>{this.props.title}:</h4>
        <div className="slider">
          {this.props.children}
        </div>
      </div>
    );
  }
}

SliderCarousel.propTypes = propTypes;
SliderCarousel.defaultProps = defaultProps;

export default SliderCarousel;
