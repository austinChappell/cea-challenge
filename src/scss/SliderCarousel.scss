.SliderCarousel {
  .slider {
    display: flex;
    overflow: auto;
    text-align: center;
    align-items: stretch;
    .item {
      min-width: 100px;
      .avatar {
        margin: 10px auto;
        border-radius: 50%;
        height: 80px;
        width: 80px;
      }
      img {
        border-radius: 50%;
      }
    }
  }
}