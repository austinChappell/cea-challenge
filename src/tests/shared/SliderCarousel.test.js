import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import SliderCarousel from '../../components/shared/SliderCarousel';

const props = {
  members: [],
};

describe('The SliderCarousel Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SliderCarousel {...props} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
