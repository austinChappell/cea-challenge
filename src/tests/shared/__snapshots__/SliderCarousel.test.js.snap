// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SliderCarousel Component should render correctly 1`] = `
<div
  className="SliderCarousel"
>
  <h4>
    :
  </h4>
  <div
    className="slider"
  />
</div>
`;
